File: 2418-sort-the-people/2418-sort-the-people.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>&nums, vector<int>&h)
    {
        int n=nums.size();
        vector<pair<int,string>>v;
        for(int i=0;i<n;i++)
        {
            v.push_back({h[i],nums[i]});
        }
        int i=0;
        sort(v.rbegin(),v.rend());
        for(auto &t:v)
        {
            nums[i++]=t.second;
        }
        return nums;
    }
};

File: 2418-sort-the-people/2418-sort-the-people.js
/**
 * @param {string[]} names
 * @param {number[]} heights
 * @return {string[]}
 */
var sortPeople = function(names, heights) {
    let nums=[];
    for(let i=0;i<names.length;i++){
        nums.push([heights[i],names[i]]);
    }
    nums.sort((a,b)=>b[0]-a[0]);
    let ans=[]
    for(let it of nums){
        ans.push(it[1]);
    }
    return ans;
};